### PR TITLE
Feature/redis index q

### PR DESCRIPTION
**Pull request by jplouis**

### CODE REVIEW

**Comment by kindkid** [comment]
You need to periodically call .requeueOldTasks() on the RedisWorkQueue. Presumably behind the same synchronization gates as the other method calls to the queue.

File: pom.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
         <version.tomcat-jdbc>7.0.27</version.tomcat-jdbc>
         <zenoss-utils.version>2.0.2</zenoss-utils.version>
         <zenoss-protocols.version>2.0.3</zenoss-protocols.version>
-        <metric-consumer.version>0.0.12</metric-consumer.version>
+        <metric-consumer.version>0.0.16</metric-consumer.version>
```
Comment (epowell) [inline pom.xml]: :+1:

**Comment by kindkid** [comment]
@jplouis could you explain the purpose of txSynchronizedQueue?

**Comment by jplouis** [comment]
@kindkid The txSynchronizedQueue is to accumulate the UUIDs to be indexed during the transaction, it also uses a set to make sure there are no duplicates. It is registered as a TransactionSynchronization so that it doesn't publish to redis until after database has been updated.

**Pull request merged**